Service/pdoShipStorage.php

```diff
@@ -53,15 +61,60 @@ public function saveShipData($newShipData)
             'INSERT INTO ship(name, weapon_power, jedi_factor, strength, team)
             VALUES(:nameVal, :weaponVal, :jediVal, :strengthVal, :teamVal)';
         $statement = $pdo->prepare($query);
-        $statement->bindParam('nameVal', $newShipData['name']);
-        $statement->bindParam('weaponVal', $newShipData['weapon_power']);
-        $statement->bindParam('jediVal', $newShipData['jedi_factor']);
-        $statement->bindParam('strengthVal', $newShipData['strength']);
-        $statement->bindParam('teamVal', $newShipData['team']);
+        $name = $newShipData->getName();
```

Comment:
I don't think you need to declare a variable for each of these. Just use the `$ship->getXYZ()` call on the `bindParam`.

Reply:
Done